sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -163,7 +163,27 @@ class SourceProgress protected[sql](
   val endOffset: String,
   val numInputRows: Long,
   val inputRowsPerSecond: Double,
-  val processedRowsPerSecond: Double) extends Serializable {
+  val processedRowsPerSecond: Double,
+  val customMetrics: String) extends Serializable {
+
+  /** SourceProgress without custom metrics. */
+  def this(
```

Comment:
I would make it `protected[sql] def this(`.

Reply:
changed.